lex.go

```diff
@@ -100,6 +162,33 @@ func tokenize(reader io.Reader, tokenCh chan NgxToken) {
 			la = "\\" + la
 		}
 
+		if token.Len() > 0 {
+			tokenStr := token.String()
+			if ext, ok := externalLexers[tokenStr]; ok {
+				if nextTokenIsDirective {
```

Comment:
Nit (micro-optimization), can this be checked first as it's the cheaper check vs looking for a key in the `externalLexers` map.

Reply:
done